src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessor.java

```diff
@@ -201,8 +219,46 @@ public void trimOldSnapshots() throws Exception {
     trimMap(shardRqMetricsMap, RQ_SNAPSHOTS);
     trimMap(httpRqMetricsMap, HTTP_RQ_SNAPSHOTS);
     trimMap(masterEventMetricsMap, MASTER_EVENT_SNAPSHOTS);
-    trimDatabases(
-        metricsDBMap, MAX_DATABASES, PluginSettings.instance().shouldCleanupMetricsDBFiles());
+
+    boolean deleteDBFiles = PluginSettings.instance().shouldCleanupMetricsDBFiles();
+    // Cleanup all but the 2 most recent metricsDB files from metricsDBMap. The most recent metricsDB files needs to be
+    // retained for future metrics query handling, the second most recent metricsDB file needs to be retained in case
```

Comment:
We have a few things going on here:
- cleaning up in-memory snapshots
- cleaning up on-disk files if batch metrics is disabled.
- re-reading config 
- And cleaning up metrics db files so that we always have a fixed number of them around.

Let's move the in-memory cleanup and on disk cleanups into separate methods of their own

Reply:
Sure